# Finding P-Value from a T-Score in Python

### In this hypoethesis test, we use a significance level of α = 0.05.

To find the p-value associated with a t-score in Python, we can use the scipy.stats.t.sf() function, which uses the following syntax:

#### scipy.stats.t.sf(abs(x), df)

where x is the t-score and df is the degrees of freedom.

### P-value in a left-tailed test:

In [22]:
# Importing the library
import scipy.stats 

# Determine the p-value
p_value = scipy.stats.t.sf(abs(-.47), df=12)


print('The p-value is : ' + str(p_value)) 

The p-value is : 0.3233906798700002


The null hypothesis is accepted because p-value > 0.05.

Note: We use "+" because we need to convert the number to string first, in order to join them together.

### P-value in the right-tailed test:

In [17]:
scipy.stats.t.sf(abs(1.87), df=24)

0.036865328383323424

The p-value is 0.037. The null hypothesis is rejected because the p-value < 0.05.

### P-value in the two-tailed test:

In [20]:
scipy.stats.t.sf(abs(1.36), df=33)*2

0.18304931466593782

The p-value is 0.183. The null hypothesis is accepted because p-value > 0.05.

# Finding P-Value from a Z-Score in Python

### Left tailed or Lower tailed test

In [23]:
scipy.stats.norm.sf(abs(-0.67))

The p-value is : 0.25142889509531013


Since p-value > 0.05, the alternative hypothesis is rejected and the null hypothesis is accepted.

### Right tailed or upper tailed test

In [24]:
scipy.stats.norm.sf(abs(1.67))

0.04745968180294733

Since p-value < 0.05, the alternative hypothesis is accepted and null hypothesis is rejected.

### Two-tailed tests

In [26]:
scipy.stats.norm.sf(abs(1.8))*2

0.07186063822585158

Since p-value > 0.05, the alternative hypothesis is rejected and the null hypothesis is accepted.